app/controllers/comments_controller.rb

```diff
@@ -77,6 +101,8 @@ def update
 
   def destroy
     if @comment.update(deleted: true)
+      @comment_thread = @comment.comment_thread
+      @comment_thread.update(reply_count: @comment_thread.comments.undeleted.size)
```

Comment:
**suggestion** (rails, code-clarity): This (and any equivalents, like L119) would be better handled by a counter_cache in the model - see Post for an example:

```ruby
counter_culture :parent, column_name: proc { |model| model.deleted? ? nil : 'answer_count' }
```

I *think* what you'd need is a similar line added to Comment:

```ruby
counter_culture :comment_thread, column_name: proc { |model| model.deleted? ? nil : 'reply_count' }
```